dart/test/keyword_test_keyword_test_generated.dart

```diff
@@ -0,0 +1,280 @@
+
+
+class ABC {
+  final int value;
+  const ABC._(this.value);
+
+  factory ABC.fromValue(int value) {
+    final result = values[value];
+    if (result == null) {
+        throw StateError('Invalid value $value for bit flag enum ABC');
+    }
+    return result;
+  }
+
+  static ABC? _createOrNull(int? value) => 
+      value == null ? null : ABC.fromValue(value);
+
+  static const int minValue = 0;
+  static const int maxValue = 2;
+  static bool containsValue(int value) => values.containsKey(value);
+
+  static const ABC void_ = ABC._(0);
+  static const ABC where = ABC._(1);
+  static const ABC stackalloc = ABC._(2);
+  static const Map<int, ABC> values = {
+    0: void_,
+    1: where,
+    2: stackalloc};
+
+  static const fb.Reader<ABC> reader = _ABCReader();
+
+  @override
+  String toString() {
+    return 'ABC{value: $value}';
+  }
+}
+
+class _ABCReader extends fb.Reader<ABC> {
+  const _ABCReader();
+
+  @override
+  int get size => 4;
+
+  @override
+  ABC read(fb.BufferContext bc, int offset) =>
+      ABC.fromValue(const fb.Int32Reader().read(bc, offset));
+}
+
+class public {
+  final int value;
+  const public._(this.value);
+
+  factory public.fromValue(int value) {
+    final result = values[value];
+    if (result == null) {
+        throw StateError('Invalid value $value for bit flag enum public');
+    }
+    return result;
+  }
+
+  static public? _createOrNull(int? value) => 
+      value == null ? null : public.fromValue(value);
+
+  static const int minValue = 0;
+  static const int maxValue = 0;
+  static bool containsValue(int value) => values.containsKey(value);
+
+  static const public NONE = public._(0);
+  static const Map<int, public> values = {
+    0: NONE};
+
+  static const fb.Reader<public> reader = _publicReader();
+
+  @override
+  String toString() {
+    return 'public{value: $value}';
+  }
+}
+
+class _publicReader extends fb.Reader<public> {
+  const _publicReader();
+
+  @override
+  int get size => 4;
+
+  @override
+  public read(fb.BufferContext bc, int offset) =>
+      public.fromValue(const fb.Int32Reader().read(bc, offset));
+}
+
+class KeywordsInUnionTypeId {
+  final int value;
+  const KeywordsInUnionTypeId._(this.value);
+
+  factory KeywordsInUnionTypeId.fromValue(int value) {
+    final result = values[value];
+    if (result == null) {
+        throw StateError('Invalid value $value for bit flag enum KeywordsInUnionTypeId');
+    }
+    return result;
+  }
+
+  static KeywordsInUnionTypeId? _createOrNull(int? value) => 
+      value == null ? null : KeywordsInUnionTypeId.fromValue(value);
+
+  static const int minValue = 0;
+  static const int maxValue = 2;
+  static bool containsValue(int value) => values.containsKey(value);
+
+  static const KeywordsInUnionTypeId NONE = KeywordsInUnionTypeId._(0);
+  static const KeywordsInUnionTypeId static_ = KeywordsInUnionTypeId._(1);
+  static const KeywordsInUnionTypeId internal = KeywordsInUnionTypeId._(2);
```

Comment:
It seems pretty weird to have no consistency here. Dart style says to not use ALLCAPS. Having an underscore at the end of the identifier is also very strange and visually confusing for my Dart sensibilities - and seems more like a C++ thing than a Dart thing.

Reply:
Yeah, I agree there isn't any consistency. This is caused by variants using `kKeep` case, which is "whatever the schema says" and the default enum is "NONE" in all caps. We can trivially change variants to `kUpperCamel` case, or whatever Dart prefers but that should be another PR, imo. I intend to keep the generated files as similar as possible for this one.